envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotProperties.kt

```diff
@@ -64,8 +64,16 @@ class IncomingPermissionsProperties {
     var sourceIpAuthentication = SourceIpAuthenticationProperties()
 }
 
+typealias ClusterName = String
+
 class SourceIpAuthenticationProperties {
     var enabledForServices: List<String> = listOf()
+    var staticIpRanges: MutableMap<ClusterName, List<IpWithPrefix>> = mutableMapOf()
+}
+
+class IpWithPrefix {
+    var ip: String = "0.0.0.0"
```

Comment:
I would change default ip to an empty string. Default values don't make sense in this class, we should always override them. So EC / Envoy should fail if we don't provide them, instead of working without warning with default values. I think the range 0.0.0.0/0 may actually match all IP addresses (I might be wrong).

I would change prefixLength to -1, for the same reason.